dwave/cloud/client/base.py

```diff
@@ -1449,9 +1453,16 @@ def _do_cancel_problems(self):
         finally:
             session.close()
 
-    def _poll(self, future):
+    def _poll(self, future: Future) -> None:
         """Enqueue a problem to poll the server for status."""
 
+        # split for simplicity
+        if self.config.polling_schedule.strategy == PollingStrategy.BACKOFF:
+            return self._poll_using_backoff(future)
+        else:
+            return self._poll_using_long_polling(future)
```

Comment:
```python
        if self.config.polling_schedule.strategy == PollingStrategy.BACKOFF:
            return self._poll_using_backoff(future)
        elif self.config.polling_schedule.strategy == PollingStrategy.LONG_POLLING:
            return self._poll_using_long_polling(future)
        else:
            raise RuntimeError("unexpected polling strategy")
```
might make future development a tiny bit easier

Reply:
Actually had it like that initially, but simplified later. But sure, can add it back, better to explicitly fail.